pygmt/src/inset.py

```diff
@@ -0,0 +1,131 @@
+        inset using the default pen; specify a different pen
+        with **+p**\ *pen*. Add **+g**\ *fill* to fill the logo box
+        [Default is no fill].
+        Append **+c**\ *clearance*  where *clearance* is either
+        *gap*, *xgap*\ /\ *ygap*, or *lgap*\ /\ *rgap*\ /\ *bgap*\ /\
+        *tgap* where these items are uniform, separate in x- and
+        y-direction, or individual side spacings between logo and border.
+        Append **+i** to draw a secondary, inner border as well. We use a
+        uniform *gap* between borders of 2\ **p** and the default pen
+        unless other values are specified. Append **+r** to draw rounded
+        rectangular borders instead, with a 6\ **p** corner radius. You
+        can override this radius by appending another value. Append
+        **+s** to draw an offset background shaded region. Here, *dx*/*dy*
+        indicates the shift relative to the foreground frame
+        [4\ **p**/-4\ **p**] and *shade* sets the fill style to use for
+        shading [Default is gray50].
+    margin : int or str or list
+        This is clearance that is added around the inside of the inset.
+        Plotting will take place within the inner region only. The margins
+        can be a single value, a pair of values separated (for setting
+        separate horizontal and vertical margins), or the full set of four
+        margins (for setting separate left, right, bottom, and top
+        margins). When passing multiple values, it can be either a list or
+        a string with the values separated by forward
+        slashes [Default is no margins].
+    no_clip : bool
+        Do NOT clip features extruding outside map inset boundaries [Default
+        will clip].
+    {V}
+
+    Examples
+    --------
+    >>> import pygmt
+    >>>
+    >>> # Create the larger figure
+    >>> fig = pygmt.Figure()
+    >>> fig.coast(region="MG+r2", water="lightblue", shorelines="thin")
+    >>> # Use a "with" statement to initialize the inset context manager
+    >>> # Setting the position to top left and a width of 3.5 centimeters
+    >>> with fig.inset(position="jTL+w3.5c+o0.2c", margin=0, box="+pgreen"):
+    ...     # Map elements under the "with" statement are plotted in the inset
+    ...     fig.coast(
+    ...         region="g",
+    ...         projection="G47/-20/?c",
+    ...         land="gray",
+    ...         water="white",
+    ...         dcw="MG+gred",
+    ...     )
+    ...
+    >>> # Map elements outside the `with` block are plotted in the main figure
+    >>> fig.logo(position="jBR+o0.2c+w3c")
+    """
+    kwargs = self._preprocess(**kwargs)  # pylint: disable=protected-access
+    with Session() as lib:
+        try:
+            lib.call_module("inset", f"begin {build_arg_string(kwargs)}")
+            yield
+        finally:
+            v_arg = build_arg_string(kwargs.fromkeys("V"))
+            lib.call_module("inset", f"end {v_arg}".strip())
```

Comment:
Can you confirm that `kwargs.fromkeys("V")` works as you expect?

I just tried:
```
with fig.inset(position="jTL+w3.5c+o0.2c", margin=0, box="+pgreen", verbose='d'):
```

`kwargs` is `{'D': 'jTL+w3.5c+o0.2c', 'F': '+pgreen', 'M': 0, 'V': 'd'}`, but `v_arg` is `{'V': None}`.

Reply:
Seems to work:

```python-traceback
___________________________________________________________________ [doctest] pygmt.src.inset.inset ____________________________________________________________________
EXAMPLE LOCATION UNKNOWN, not showing all tests of that example
??? >>> with fig.inset(position="jTL+w3.5c+o0.2c", margin=0, box="+pgreen", verbose="d"):
??? ...     # Map elements under the "with" statement are plotted in the inset
??? ...     fig.coast(
??? ...         region="g",
??? ...         projection="G47/-20/3.5c",
??? ...         land="gray",
??? ...         water="white",
??? ...         dcw="MG+gred",
??? ...     )
Expected nothing
Got:
    {'D': 'jTL+w3.5c+o0.2c', 'F': '+pgreen', 'M': 0, 'V': 'd'}
    {'D': 'jTL+w3.5c+o0.2c', 'F': '+pgreen', 'M': 0, 'V': 'd'} 

/home/username/Documents/github/pygmt/pygmt/src/inset.py:None: DocTestFailure
------------------------------------------------------------------------- Captured stderr call -------------------------------------------------------------------------
inset [DEBUG]: Map distance calculation will be using great circle approximation with authalic auxiliary latitudes and authalic (R_2) radius = 6371007.1809 m, in meter.
inset [DEBUG]: Option -- option: Modifier +r unrecognized
inset [DEBUG]: Filename has invalid modifiers - probably not a file with modifiers (MG+r2)
inset [DEBUG]: Look for file MG+r2 in /home/username/.gmt
inset [DEBUG]: Look for file MG+r2 in /home/username/.gmt/cache
inset [DEBUG]: Look for file MG+r2 in /home/username/.gmt/server
inset [DEBUG]: Got country code for region (MG+r2)
inset [DEBUG]: DCW: Found 250 countries, 8 countries with states, and 281 states
inset [DEBUG]: Requested 1 DCW items: MG
inset [INFORMATION]: Using country and state data from dcw-gmt
inset [INFORMATION]: Title  : DCW-GMT - The Digital Chart of the World for the Generic Mapping Tools
inset [INFORMATION]: Source : Processed by the GMT Team, 2018-JUL-01
inset [INFORMATION]: Version: 1.1.4
inset [INFORMATION]: Madagascar
inset [DEBUG]: Longitude range initialized to 43.1914/50.4838
inset [INFORMATION]: Region implied by DCW polygons is 42/52/-26/-10
inset [DEBUG]: Map distance calculation will be using great circle approximation with authalic auxiliary latitudes and authalic (R_2) radius = 6371007.1809 m, in meter.
inset [DEBUG]: Anchor point specified via justification code: TL
inset [DEBUG]: Map inset attributes: justify = 9, dx = 0.0787402 dy = 0.0787402
inset [INFORMATION]: Spherical approximation used
inset [INFORMATION]: Central meridian not given, default to 47
inset [DEBUG]: Projected values in meters: -555975 555975 -2.89107e+06 -1.11195e+06
inset [INFORMATION]: Map scale is 118.608 km per cm or 1:1.18608e+07.
inset [DEBUG]: Use PS filename /home/username/.gmt/sessions/gmt_session.6034/gmt_1.ps-
inset [DEBUG]: Running in PS mode modern
inset [DEBUG]: Use PS filename /home/username/.gmt/sessions/gmt_session.6034/gmt_1.ps-
inset [DEBUG]: Append to hidden PS file /home/username/.gmt/sessions/gmt_session.6034/gmt_1.ps-
inset [DEBUG]: Got session name as pygmt-session and default graphics formats as pdf
inset [DEBUG]: Converted code 9 to i = 1, j = 2 and finally x = 0 and y = 5.90551
inset [DEBUG]: Convert code inside reference point coordinates from justification LT to 0, 0
inset [DEBUG]: Before justify = 9, Dim x = 1.37795 y = 1.37795, Reference x = 0 y = 5.90551
inset [DEBUG]: After justify = 9, Offset x = 0.0787402 y = 0.0787402, Reference x = 0 y = 4.52756
inset [DEBUG]: After shifts, Reference x = 0.0787402 y = 4.44882
inset [INFORMATION]: Map inset lower left corner and dimensions (in cm): 0.2 11.3 3.5 3.5
inset [INFORMATION]: Place rectangular back panel
inset [DEBUG]: Clearance: 0/0/0/0
inset [INFORMATION]: No file /home/username/.gmt/sessions/gmt_session.6034/gmt.frame with frame information - no adjustments made
inset [DEBUG]: inset: Wrote inset settings to information file /home/username/.gmt/sessions/gmt_session.6034/gmt.inset.1
inset [DEBUG]: Current size of half-baked PS file /home/username/.gmt/sessions/gmt_session.6034/gmt_1.ps- = 55568.
inset (gmtlib_free_tmp_arrays): tried to free unallocated memory
```

Reply:
I think what you see all comes from **gmt inset begin -Vd**, not **gmt inset end -Vd**.

Reply:
Ah true, silly mistake. Use this instead:

```suggestion
            v_arg = build_arg_string({"V": kwargs.get("V")})
            lib.call_module("inset", f"end {v_arg}".strip())
```